### ISSUE TITLE
Implicit Section Linking Based on Antecedent

### ISSUE THREAD

**Issue by brandon-carag** [criticaljuncture/reference_parser#39]
### Description
We received the following user report:

```
INCORRECT HYPERLINKS

10 CFR § 71.5(a)(1)(ii) states:

Marking and labeling - 49 CFR part 172: subpart D; and §§ 172.400 through 172.407 and §§ 172.436 through 172.441 of subpart E.

Except for the first hyperlink, all of the hyperlinks erroneously link to title 10 instead of title 49.
```

![Screen Shot 2023-02-03 at 2 00 27 PM](https://user-images.githubusercontent.com/3423367/216719140-15e096eb-4fc4-42f7-a460-891c9421cfa2.png)

https://www.ecfr.gov/current/title-10/chapter-I/part-71#p-71.5(a)(1)(iii)

### Expected behavior
Not sure there's a great universal way to handle this scenario or take into account context based on the current logic, but mentioning in case.  Feel free to ignore if deemed insignificant.

**Comment by peregrinator** [comment]
There's a few different edge cases in the whole of 71.5 that are worth looking at. https://www.ecfr.gov/current/title-10/chapter-I/part-71/subpart-A/section-71.5

@damon I think the priority is to not link to Title 10 in these cases - great if you can get it to work reliably but also fine if we want to consider these as ambiguous and just not link

**Comment by rushed** [comment]
resolved via 6655996b60584f83e1892000d6ebca4267fdaeb5
